classes/class-wc-connect-notice-manager.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+// No direct access please
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'WC_Connect_Notice_Manager' ) ) {
+
+	class WC_Connect_Notice_Manager {
+		/**
+		 * @var WC_Connect_Service_Schemas_Store
+		 */
+		private $schemas_store;
+
+		public function __construct( WC_Connect_Service_Schemas_Store $schemas_store ) {
+			$this->schemas_store = $schemas_store;
+		}
+
+		/**
+		 * Returns if a notice ID was already dismissed in the past.
+		 *
+		 * @param string $id Notice ID.
+		 *
+		 * @return bool Is the notice ID marked as dismissed in the past.
+		 */
+		public function is_dismissed( $id ) {
+			$dismissed_in_transient = get_transient( 'wcc_notice_dismissed_' . $id ); // Legacy approach.
```

Comment:
Up until now the extension would store the fact of someone dismissing a notice in a transient. This had two downsides:

1. A transient would only live for up to a specific time, meaning the notice would reappear.
2. Transients by their nature expire after a given time _or earlier_, potentially causing the notice to be displayed again.

I have created this follow-up issue to remove the support for this in the future: https://github.com/Automattic/woocommerce-services/issues/2641